internal/ansible/runner/runner.go

```diff
@@ -277,6 +277,11 @@ func (r *runner) Run(ident string, u *unstructured.Unstructured, kubeconfig stri
 		// link the current run to the `latest` directory under artifacts
 		currentRun := filepath.Join(inputDir.Path, "artifacts", ident)
 		latestArtifacts := filepath.Join(inputDir.Path, "artifacts", "latest")
+		// verify if artifacts directory exists and error out if it does not exist
+		if _, err := os.Stat(latestArtifacts); os.IsNotExist(err) {
```

Comment:
The symlink should be attempted only if `latestArtifacts` does not exist, or it did and `os.Remove(latestArtifacts)` succeeded. This should look something like
```go
		if _, err = os.Lstat(latestArtifacts); err != nil {
			if !errors.Is(err, os.ErrNotExist) {
				logger.Error(err, "Latest artifacts dir")
				return
			}
		} else if err = os.Remove(latestArtifacts); err != nil {
			logger.Error(err, "Error removing the latest artifacts symlink")
			return
		}

		if err = os.Symlink(currentRun, latestArtifacts); err != nil {
			logger.Error(err, "Error symlinking latest artifacts")
		}
```